### PR TITLE
Using hypercore-streams internally

### PR DESCRIPTION
**Pull request by martinheidegger**
This PR internalizes hypercore-streams and uses streamx consistently.

As `streamx` is not 100% compatible with `readable-streams`, should this be considered a breaking change?

The API did not change but in the tests I needed to add an delay, as `streamx` doesn't immediately create a batch.

~~_It is currently a Draft PR: it requires for several merges:_~~

- [x] https://github.com/mafintosh/merkle-tree-stream/pull/7
- [x] https://github.com/mafintosh/hypercore-streams/pull/2
- [x] https://github.com/mafintosh/hypercore-streams/pull/3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,30 +7,31 @@
...
     "last-one-wins": "^1.0.4",
     "memory-pager": "^1.0.2",
-    "merkle-tree-stream": "^3.0.3",
+    "merkle-tree-stream": "^4.0.0",
     "nanoguard": "^1.2.0",
     "nanoresource": "^1.3.0",
     "pretty-hash": "^1.0.1",
     "random-access-file": "^2.1.0",
     "sparse-bitfield": "^3.0.0",
+    "streamx": "^2.7.1",
```
Comment (mafintosh) [inline package.json:27]: i don't think we use this directly now?

File: package.json [hunk 2]
```diff
@@ -7,30 +7,31 @@
...
     "timeout-refresh": "^1.0.3",
     "uint64be": "^2.0.1",
     "unordered-array-remove": "^1.0.2",
     "unordered-set": "^2.0.0"
   },
   "devDependencies": {
+    "buffer-from": "^1.1.1",
```
Comment (mafintosh) [inline package.json:34]: needed?

File: test/basic.js [hunk 3]
```diff
@@ -445,15 +445,17 @@ tape('writes are batched', function (t) {
...
+    ws.write('cd')
+    ws.write('ef')
+    ws.end(function () {
+      t.deepEquals(trackingRam.log.data, [
+        { write: [0, Buffer.from('ab')] },
+        { write: [2, Buffer.from('cdef')] }
+      ])
+      feed.close(function () {
+        t.end()
+      })
```
Comment (mafintosh) [inline test/basic.js:458]: would rather change the result of the assert, ie that only one write is performed and then not have the setImmediate.

thats a cool new feature we get from streamx :)

File: test/streams.js [hunk 4]
```diff
@@ -119,7 +119,9 @@ function test (batch = 1) {
         t.fail('should never end')
       })
 
-      feed.append(['c', 'd', 'e'])
+      setImmediate(function () {
+        feed.append(['c', 'd', 'e'])
+      })
```
Comment (mafintosh) [inline test/streams.js:124]: why is this needed?

File: test/basic.js (hunk 3 above)
Comment (martinheidegger) [reply test/basic.js:458]: Hmm, i think its a good idea to test blocking still (if eventually a block will be formed), is it a good idea to have a test that has two instantly and two after the setImmediate?

File: package.json (hunk 1 above)
Comment (martinheidegger) [reply package.json:27]: Will fix, thanks for pointing out!

File: package.json (hunk 2 above)
Comment (martinheidegger) [reply package.json:34]: Yes, its needed for the tests. Previously it got installed with one of the removed libraries: https://github.com/hypercore-protocol/hypercore/blob/21e9e666bc182594172c4eb746cfbad6ffdb5ee8/test/streams.js#L4

Comment (mafintosh) [reply package.json:34]: Nice, makes sense. Yea we should just move it to Buffer.from at some point

File: test/basic.js (hunk 3 above)
Comment (mafintosh) [reply test/basic.js:458]: yea that makes sense

File: test/streams.js (hunk 4 above)
Comment (martinheidegger) [reply test/streams.js:124]: Same reason as in https://github.com/hypercore-protocol/hypercore/pull/276#discussion_r514252973

File: test/basic.js (hunk 3 above)
Comment (martinheidegger) [reply test/basic.js:459]: fixed in https://github.com/hypercore-protocol/hypercore/pull/276/commits/875889d140848a090f8f8bbe199672f1481d77d2

File: package.json (hunk 1 above)
Comment (martinheidegger) [reply package.json]: fixed in https://github.com/hypercore-protocol/hypercore/pull/276/commits/875889d140848a090f8f8bbe199672f1481d77d2

**Comment by martinheidegger** [comment]
@mafintosh I addressed the open issues in https://github.com/hypercore-protocol/hypercore/pull/276/commits/875889d140848a090f8f8bbe199672f1481d77d2 - can I improve it somehow?

**Comment by mafintosh** [comment]
nope, all good from me

**Comment by mafintosh** [comment]
9.6.0
